### ISSUE TITLE
⚠️ Updated! Instafeed v1 will stop working June 29, 2020 ⚠️

### ISSUE THREAD

**Issue by benjamin-hull** [stevenschobert/instafeed.js#640]
**Updated: March 31, 2020**. The legacy API shutdown will now happen on June 29, 2020. Please update instafeed.js to version 2 before then. New integrations with the legacy API are still not possible - for new integrations please start with instafeed.js v2.

## If you have problems getting instafeed.js to work, or it has stopped working on your site, please do not open an issue.

The legacy API is in the process of being retired, which **will cause instafeed.js to stop working on ~~March 31st, 2020~~ June 29, 2020**. Instagram is no longer issuing access tokens for the legacy API, so you will not be able to start using instafeed.js even before this date.

For more information, please see the project [README](https://github.com/stevenschobert/instafeed.js/blob/master/README.md) and issue #345

We are currently looking into ways to make instafeed.js work with Instagram's new API, but we don't know if or when that might be available. To keep up to date with that discussion, please see issue #635.

**Comment by paolovador** [comment]
FYI [there's a new update](https://developers.facebook.com/blog/post/2020/01/14/instagram-basic-display-api-long-lived-access-tokens-available/)

"February 20, 2020 Update: To provide you with more time to make changes, we will now be shutting down the Instagram Legacy API Platform on March 31, 2020 instead of March 2, 2020 as originally communicated."

You can change the dates on your issues and Readme

**Comment by jhvanderschee** [comment]
I have created/found a dead simple alternative way to show your latest instagram pics on your website. See: https://jekyllcodex.org/without-plugin/instagram/

**Comment by ryangjchandler** [comment]
Having to use Zapier seems a bit overkill. I'm currently working on a JavaScript library that will provide a JavaScript feed without the legacy API, or any API token for that fact. It will be dependency free and super lightweight.

**Comment by jhvanderschee** [comment]
Although I agree with you about the overkill, I want to stress that:

1. My solution works
1. My solution uses just 11 lines of code and 3 lines of CSS
1. My solution will not be blocked by Instagram, due to unintended use

I am very curious what you have in store for us.

**Comment by ryangjchandler** [comment]
> Although I agree with you about the overkill, I want to stress that:
> 
> 1. My solution works
> 2. My solution uses just 11 lines of code and 3 lines of CSS
> 3. My solution will not be blocked by Instagram, due to unintended use
> 
> I am very curious what you have in store for us.

Gave up on my library, found this one instead: https://github.com/jsanahuja/InstagramFeed

Already using on a client site, works quite well & maintenance seems pretty active.

**Comment by jhvanderschee** [comment]
I like it. My only concern: Do you think that scraping the Instagram page is reliable? One altered CORS header and the solution is 'no more'.

**Comment by jsanahuja** [comment]
> I like it. However, I still have some concerns:
> 
> * Do you think that scraping the Instagram page is reliable? One altered CORS header and the solution is 'no more'.
> * What about the rate limit on these pages? See: [jsanahuja/jquery.instagramFeed#54 (comment)](https://github.com/jsanahuja/jquery.instagramFeed/issues/54#issuecomment-594175869)

As long as scraping is being done client-side the rate limit only aplies to single users. As far as I know, that rate limit is the same you would have visiting Instagram with your browser.

> > Although I agree with you about the overkill, I want to stress that:
> > 
> > 1. My solution works
> > 2. My solution uses just 11 lines of code and 3 lines of CSS
> > 3. My solution will not be blocked by Instagram, due to unintended use
> > 
> > I am very curious what you have in store for us.
> 
> Gave up on my library, found this one instead: https://github.com/jsanahuja/InstagramFeed
> 
> Already using on a client site, works quite well & maintenance seems pretty active.

Thank you :)

**Comment by jhvanderschee** [comment]
Great! I love it! Thank you for your answers.

**Comment by mader80** [comment]
Hey guys. I tried using https://github.com/jsanahuja/InstagramFeed. Getting the following error when trying to use it:

VM140 InstagramFeed.js:218 Uncaught TypeError: Cannot set property 'innerHTML' of null
    at Object.display (VM140 InstagramFeed.js:218)
    at VM140 InstagramFeed.js:226
    at XMLHttpRequest.xhr.onload (VM140 InstagramFeed.js:96)

Here's my implementation:

	<script src="js/InstagramFeed.js"></script>
	<script>
	    (function(){
	        new InstagramFeed({
	            'username': 'screentek',
	            'container': document.getElementById("instagram-feed"),
	            'display_profile': true,
	            'display_biography': true,
	            'display_gallery': true,
	            'callback': null,
	            'styling': true,
	            'items': 8,
	            'items_per_row': 4,
	            'margin': 1 
	        });
	    })();
	</script>

And of course the div:
<div id="instagram-feed"></div>


Did you guys run into this? Seems like the InstagramFeed.js just doesn't work or has issues...

**Comment by jhvanderschee** [comment]
Your object is empty... 

```
<div id="instagram-feed"></div>
<script src="js/InstagramFeed.js"></script>
<script>
    (function(){
        new InstagramFeed({
            'username': 'screentek',
            'container': document.getElementById("instagram-feed"),
            'display_profile': true,
            'display_biography': true,
            'display_gallery': true,
            'callback': null,
            'styling': true,
            'items': 8,
            'items_per_row': 4,
            'margin': 1 
        });
    })();
</script>
```

**Comment by mader80** [comment]
You are right! Well technically I did not forget it, but I put the script above the div. Now I've got the div above the script and it works. Thanks!!!

**Comment by benjamin-hull** [comment]
Head-up folks: A new version of instafeed.js powered by the new Basic Display API is incoming. Please see https://github.com/stevenschobert/instafeed.js/issues/643 for details as they're available - and thank you very much for your patience!

**Comment by benjamin-hull** [comment]
A further note: I'm going to clean up some of the comments here that relate to implementation questions on other libraries. Hope that doesn't cause any inconvenience - thank you.

**Comment by Luke-SF** [comment]
I see that this is no longer on the front page. Is that because this was postponed another 90 days? https://www.instagram.com/developer/

**Comment by benjamin-hull** [comment]
@Luke-SF It's not because of that, no - only because I didn't realize this had happened until you posted the link. We did ask them to consider a further extension though - so that's great news!

The reason I unpinned this issue is because instafeed v2 is available (though in its very early stages), and that uses the Basic Display API. If you're ready, you can start looking at the migration guide in the project wiki to upgrade to the new version.

**Comment by darcemac** [comment]
> I have created/found a dead simple way to show your latest instagram pics on your website. Please read about it here: https://jekyllcodex.org/blog/instafeed-js-alternative-for-instagram/. Note that using Jekyll is not a requirement.
> 
> Basically the code is:
> 
> ```
> <p id="instafeed"></p>
> 
> <script src="/js/jquery.min.js"></script>
> <script type="text/javascript">
> $.get('https://zapier.com/engine/rss/2502510/jhvanderschee', function (data) {
>     $(data).find("item").each(function () { // or "item" or whatever suits your feed
>         var el = $(this);
>         var title = el.find("title").text();
>         var link = el.find("link").text();
>         var image = el.find("enclosure").attr('url');
>         var description = el.find("description").text();
>         $('#instafeed').append('<a href="'+encodeURI(link)+'" target="_blank" title="'+title.replace('Caption: ','')+'"><img src="'+encodeURI(image)+'" alt="'+title.replace('Caption: ','')+'" /></a>');
>     });
> });
> </script>
> 
> <style>
>     #instafeed {overflow: auto; margin-left: -1%;}
>     #instafeed a {float: left; display: block; margin: 0 0 1% 1%; width: 19%;}
>     #instafeed a img {width: 100%;}
> </style>
> ```

**Comment by darcemac** [comment]
Great alternative solution. Is it possible to limit how many items get displayed?

**Comment by benjamin-hull** [comment]
Hi @darcemac - you're probably better off asking questions about @jhvanderschee's solution on their blog, rather than here. I'll probably clean up these comments in a couple of days just to keep things tiday.

Thanks!

**Comment by jhvanderschee** [comment]
@darcemac Yes that is possible. You will find the code on JekyllCodex.org.

**Comment by jhvanderschee** [comment]
Sadly Instagram seems to have killed my alternative solutions by constantly changing the URL through a URL signature that expires. Zapier no more...

**Comment by stevenschobert** [comment]
Closing out this issue since v2 is officially released: https://github.com/stevenschobert/instafeed.js/issues/643#issuecomment-851607996

Please open up new issues going forward!
